### PR TITLE
reference type fix

### PR DESCRIPTION
**Pull request by Hentioe**
# What does this PR do?

Fixed a bug where the `type` parameter of the `CreateTable#reference` method was invalid

# Any background context you want to provide?

What is happening now is that, regardless of the value of the `type` parameter, the resulting SQL will always be of type integer.

This PR's commits is based on the `v0.8.4` tag, not the master branch. Because the master branch currently has compilation errors.

### CODE REVIEW

File: src/jennifer/migration/table_builder/create_table.cr [hunk 1]
```diff
@@ -93,7 +93,7 @@ module Jennifer
           column = Inflector.foreign_key(name)
           is_null = options.has_key?(:null) ? options[:null] : true
 
-          integer(column, { :type => type, :null => is_null })
+          field(column, type, { :null => is_null })
```
Comment (imdrasil) [inline src/jennifer/migration/table_builder/create_table.cr]: in common case you would like to use "internal types". To achieve this and also allow developer to pass native type if required suggest 

```crystal
field_internal_type = options.has_key?(:sql_type) ? nil : type
@fields[name.to_s] = build_column_options(field_internal_type, options)
```

**Comment by imdrasil** [comment]
Also what issues do you have with master branch?

File: src/jennifer/migration/table_builder/create_table.cr (hunk 1 above)
Comment (Hentioe) [reply src/jennifer/migration/table_builder/create_table.cr]: Yes. I modified it.

**Comment by Hentioe** [comment]
@imdrasil There was no error in the master branch, which was caused by my not upgrading `jennifer sqlite3_adapter`.

**Comment by Hentioe** [comment]
If I add the `:sql_type` options parameter, the `type` parameter is meaningless.

Maybe this can be avoided?

``` crystal
t.reference :foo, :text, {:sql_type => true}
```

This is just my personal suggestion.

**Comment by imdrasil** [comment]
I've got your point. Will take this into consideration

**Pull request merged**